ts/features/payments/common/components/__tests__/PaymentCard.test.tsx

```diff
@@ -57,3 +65,107 @@ describe("PaymentCard", () => {
+    expect(component).toMatchSnapshot();
+  });
+
+  it(`should render the Pressable component when passed an OnPress`, () => {
+    const handler = () => null;
+    const component = renderCardSmall({
+      cardType: "PAYPAL",
+      onCardPress: handler,
+      testID
+    });
+    expect(component.queryByTestId(`${testID}-pressable`)).not.toBeNull();
+  });
+
+  it(`should render an error icon in case of error `, () => {
+    const component = renderCardSmall({
+      cardType: "PAYPAL",
+      isError: true,
+      testID
+    });
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId(`${testID}-errorIcon`)).not.toBeNull();
+  });
+  it(`should not render a pressable if no handler is passed`, () => {
+    const component = renderCardSmall({
+      cardType: "PAYPAL",
+      isError: true,
+      testID
+    });
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId(`${testID}-pressable`)).toBeNull();
+  });
+  it(`should render a skeleton if loading`, () => {
+    const component = renderCardSmall({
+      isLoading: true,
+      testID
+    });
+    expect(component.queryByTestId(`${testID}-skeleton`)).not.toBeNull();
+  });
+});
+
+describe("PaymentCardBig component", () => {
+  const testID = "PaymentCardBigTestID";
+  jest.useFakeTimers();
+  it(`matches snapshot for loading`, () => {
+    const component = renderCardBig({ isLoading: true });
+    expect(component).toMatchSnapshot();
+  });
+  it(`matches snapshot for paypal`, () => {
+    const component = renderCardBig({
+      cardType: "PAYPAL",
+      holderEmail: "someEmail@test.com"
+    });
+    expect(component).toMatchSnapshot();
+  });
+
+  it(`should render a phone number in case of BancomatPay`, () => {
+    const component = renderCardBig({
+      cardType: "BANCOMATPAY",
+      phoneNumber: "1234567890",
+      holderName: "holderName",
+      testID
+    });
+    expect(component).not.toBeNull();
+    expect(component.queryByText("1234567890")).not.toBeNull();
+  });
+  it(`should render a skeleton when loading`, () => {
+    const component = renderCardBig({
+      isLoading: true,
+      testID
+    });
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId(`${testID}-skeleton`)).not.toBeNull();
+  });
+});
+
+function renderCardBig(props: PaymentCardBigProps) {
+  const globalState = appReducer(undefined, applicationChangeState("active"));
+  return renderScreenWithNavigationStoreContext<GlobalState>(
+    () => <PaymentCardBig {...props} />,
+    ROUTES.WALLET_HOME,
+    {},
+    createStore(appReducer, globalState as any)
+  );
+}
+
+function renderCardSmall(props: PaymentCardSmallProps) {
+  const globalState = appReducer(undefined, applicationChangeState("active"));
+  return renderScreenWithNavigationStoreContext<GlobalState>(
+    () => <PaymentCardSmall {...props} />,
+    ROUTES.WALLET_HOME,
+    {},
+    createStore(appReducer, globalState as any)
+  );
+}
```

Comment:
Imho since we have `PaymentCard`, `--Small`,`--Big`,`--Carousel` I was thinking that either we have a single file for each and every component, thus having a lot of duplicated boilerplate, or we just merge it all in a single, not enormous of course, file -- and the second option seemed like the most human one of the two. What are the upsides of moving it all in different files other than "cleanliness" at the cost of file tree cluttering?